src/benchmarkstt/api/jsonrpc.py

```diff
@@ -72,16 +72,21 @@ def _(*args, **kwargs):
         # copy signature from original
         sig = signature(cls)
 
+        def param_filter(param):
+            return param.kind not in (Parameter.VAR_KEYWORD, Parameter.VAR_POSITIONAL)
+
         cb_params = signature(callback).parameters.values()
+        cb_params = list(filter(param_filter, cb_params))
+
         extra_params = [parameter for parameter in cb_params
-                        if parameter.name != 'cls' and
-                        parameter.kind not in (Parameter.VAR_KEYWORD,
-                                               Parameter.VAR_POSITIONAL)]
+                        if parameter.name != 'cls']
+
         if len(extra_params):
             params = list(filter(lambda x: x.default is _empty, extra_params))
-            params.extend(sig.parameters.values())
+            params.extend(filter(param_filter, sig.parameters.values()))
```

Comment:
I'm not familiar with the JSON RPC side of things, but on this line you've added a `filter` instead of a `list`, like the line below. Is this going to be problematic?

Reply:
When you extend a list with a filter (which returns a generator), the generator gets fully iterated and the result is the list extended with the results from that generator. So it stays a list...